src/app/flows/fines/fines-acc/services/utils/fines-acc-payload-transform-defendant-data.utils.ts

```diff
@@ -78,40 +180,32 @@ export const transformDefendantAccountPartyPayload = (
-    facc_party_add_amend_convert_contact_telephone_number_mobile: contact_details?.mobile_telephone_number || null,
-    facc_party_add_amend_convert_contact_telephone_number_home: contact_details?.home_telephone_number || null,
-    facc_party_add_amend_convert_contact_telephone_number_business: contact_details?.work_telephone_number || null,
-    facc_party_add_amend_convert_vehicle_make: vehicle_details?.vehicle_make_and_model || null,
-    facc_party_add_amend_convert_vehicle_registration_mark: vehicle_details?.vehicle_registration || null,
-    facc_party_add_amend_convert_language_preferences_document_language:
-      language_preferences?.document_language_preference?.language_code || null,
-    facc_party_add_amend_convert_language_preferences_hearing_language:
-      language_preferences?.hearing_language_preference?.language_code || null,
-    facc_party_add_amend_convert_employer_company_name: employer_details?.employer_name || null,
-    facc_party_add_amend_convert_employer_reference: employer_details?.employer_reference || null,
-    facc_party_add_amend_convert_employer_email_address: employer_details?.employer_email_address || null,
-    facc_party_add_amend_convert_employer_telephone_number: employer_details?.employer_telephone_number || null,
-    facc_party_add_amend_convert_employer_address_line_1: employer_details?.employer_address?.address_line_1 || null,
-    facc_party_add_amend_convert_employer_address_line_2: employer_details?.employer_address?.address_line_2 || null,
-    facc_party_add_amend_convert_employer_address_line_3: employer_details?.employer_address?.address_line_3 || null,
-    facc_party_add_amend_convert_employer_address_line_4: employer_details?.employer_address?.address_line_4 || null,
-    facc_party_add_amend_convert_employer_address_line_5: employer_details?.employer_address?.address_line_5 || null,
-    facc_party_add_amend_convert_employer_post_code: employer_details?.employer_address?.postcode || null,
-  };
+  const isCompany = partyType === 'company';
+  const isIndividualOrParentGuardian = partyType === 'individual' || partyType === 'parentGuardian';
+
+  // Create base state with common fields
+  const baseState = createBaseState(address, contact_details, vehicle_details, language_preferences);
+
+  // If partyType is specified, prioritize it over organisation_flag
+  if (isCompany) {
+    return getCompanyParty(baseState, organisationDetails, organisationAliases, hasAliases);
+  } else if (isIndividualOrParentGuardian) {
+    return getIndividualOrParentGuardianParty(
+      baseState,
+      individualDetails,
+      individualAliases,
+      hasAliases,
+      employer_details,
+    );
+  } else if (organisation_flag) {
+    return getCompanyParty(baseState, organisationDetails, organisationAliases, hasAliases);
+  }
+
+  return getIndividualOrParentGuardianParty(
+    baseState,
+    individualDetails,
+    individualAliases,
+    hasAliases,
+    employer_details,
+  );
```

Comment:
I don't know why you need a separate if statement to run the same thing. Also, if it is an if else condition then you wouldn't need the fallback return at the end. Furthermore, if the fallback is required then I'm not sure what it should be returning as I'm not sure what it should even display on the screen

```suggestion
  if (isCompany || organisation_flag) {
    return getCompanyParty(baseState, organisationDetails, organisationAliases, hasAliases);
  } else (isIndividualOrParentGuardian) {
    return getIndividualOrParentGuardianParty(
      baseState,
      individualDetails,
      individualAliases,
      hasAliases,
      employer_details,
    );
  } 
```